ddtrace/tracer/option.go

```diff
@@ -246,6 +250,54 @@ func newConfig(opts ...StartOption) *config {
 	return c
 }
 
+// defaultHTTPClient returns the default http.Client to start the tracer with.
+func defaultHTTPClient() *http.Client {
+	if _, err := os.Stat(defaultSocketAPM); err == nil {
+		// we have the UDS socket file, use it
+		return udsClient(defaultSocketAPM)
+	}
+	return defaultClient
+}
+
+// udsClient returns a new http.Client which connects using the given UDS socket path.
+func udsClient(socketPath string) *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socketPath)
+			},
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		},
+		Timeout: defaultHTTPTimeout,
+	}
+}
+
+// defaultDogstatsdAddr returns the default connection address for Dogstatsd.
+func defaultDogstatsdAddr() string {
+	envHost, envPort := os.Getenv("DD_AGENT_HOST"), os.Getenv("DD_DOGSTATSD_PORT")
+	host, port := "localhost", "8125"
+	if v := envHost; v != "" {
+		host = v
+	}
+	if v := envPort; v != "" {
+		port = v
+	}
+	if envHost != "" || envPort != "" {
+		// the user specified his choice via env. vars.
+		return net.JoinHostPort(host, port)
+	}
+	if _, err := os.Stat(defaultSocketDSD); err == nil {
+		// we have a socket file, use it
+		return "unix://" + defaultSocketDSD
+	}
+	// no choice, nor socket file available, default to localhost:8125
+	return "localhost:8125"
+}
```

Comment:
NIT: I think this function can be simplified a bit (tested to pass `TestDefaultDogstatsdAddr`)

```diff
diff --git a/ddtrace/tracer/option.go b/ddtrace/tracer/option.go
index 2586825..1bc8542 100644
--- a/ddtrace/tracer/option.go
+++ b/ddtrace/tracer/option.go
@@ -279,23 +279,18 @@ func udsClient(socketPath string) *http.Client {
 // defaultDogstatsdAddr returns the default connection address for Dogstatsd.
 func defaultDogstatsdAddr() string {
 	envHost, envPort := os.Getenv("DD_AGENT_HOST"), os.Getenv("DD_DOGSTATSD_PORT")
-	host, port := "localhost", "8125"
-	if v := envHost; v != "" {
-		host = v
-	}
-	if v := envPort; v != "" {
-		port = v
+	if _, err := os.Stat(defaultSocketDSD); err == nil && envHost == "" && envPort == "" {
+		// socket exists and user didn't specify otherwise via env vars
+		return "unix://" + defaultSocketDSD
 	}
-	if envHost != "" || envPort != "" {
-		// the user specified his choice via env. vars.
-		return net.JoinHostPort(host, port)
+	host, port := "localhost", "8125"
+	if envHost != "" {
+		host = envHost
 	}
-	if _, err := os.Stat(defaultSocketDSD); err == nil {
-		// we have a socket file, use it
-		return "unix://" + defaultSocketDSD
+	if envPort != "" {
+		port = envPort
 	}
-	// no choice, nor socket file available, default to localhost:8125
-	return "localhost:8125"
+	return net.JoinHostPort(host, port)
 }
 
 func statsTags(c *config) []string {
```